hpOneView/storage.py

```diff
@@ -99,53 +252,308 @@ def add_storage_pool(self, name, storageSystemUri, blocking=True,
+        """Gets a volume attachment path by id.
+
+        :param attachment_id: ID of specific volume attachment
+        :param path_id: ID of specific path in volume attachment
+        """
+        url = make_url(uri['attachments'], attachment_id, 'paths', path_id)
+
+        body = self._con.get(url)
+        return body
+
+    def get_attachments_volumes_by_id(self, attachment_id):
+        """Gets a volume attachment by id.
+
+        :param attachment_id: ID of specific volume attachment
+        """
+        url = make_url(uri['attachments'], attachment_id)
+
+        body = self._con.get(url)
+        return body
+
+    ###########################################################################
+    # Storage Volume Templates: resource models a set of parameters to use for volume creation.
+    ###########################################################################
+
+    def get_storage_volume_templates(self, filter=''):
+        """Gets a list of storage volume templates.
+
+        Examples:
+            1 - Get all storage volume templates (default)
+
+                filter = ?start=0&count=-1
+
+            2 - Get storage volume template filtering by name
+
+                filter = ?filter=name='MyTemplate'
+
+        :param filter: string with optional parameters to apply filtering and sorting operation
+        """
+        url = make_url(uri['vol-templates'], filter=filter)
+
+        body = get_members(self._con.get(url))
+        return body
+
+    def add_storage_volume_template(self, name, capacity, sto_pool_uri, shareable, provision_type, description='',
+                                    sto_system_uri=None, snapshot_pool_uri=None, blocking=True, verbose=False):
+        """Creates a new storage volume template.
+
+        :param name: name for the storage volume template resource
+        :param capacity: capacity requested for the volume in GiB
+        :param sto_pool_uri: URI of storage pool from which the volume to be created
+        :param shareable: describes if the volume is shareable (public) or private
+        :param provision_type: provisioning type of the volume - Thin or Full
+        :param description: description of the storage volume template resource
+        :param sto_system_uri: URI of storage system which contains the storage volume template
+        :param snapshot_pool_uri: URI of storage pool from which the snapshots to be created
+
+        :param blocking
+        :param verbose:
+        """
+        request = make_storage_volume_template_v3(name, capacity, sto_pool_uri, shareable, provision_type,
+                                                  description, sto_system_uri, snapshot_pool_uri)
+
+        # Temporally modify the headers passed for POST and DELETE on storage volume
+        # templates in order to work around a bug. Without these headers the call
+        # cause a NullPointerException on the appliance and a 400 gets returned.
         ori_headers = self._con._headers
```

Comment:
Do you happen to know if this still needs to be done in the headers for v200 calls?